confluent_kafka/schema_registry/serdes.py

```diff
@@ -0,0 +1,264 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import io
+import struct
+
+from fastavro import schemaless_writer, schemaless_reader
+
+from .schema_registry_client import SchemaRegistryClient
+from confluent_kafka.serialization.error import SerializationError
+from confluent_kafka.serialization.kafka_serdes import Serializer, Deserializer
+
+_MAGIC_BYTE = 0
+
+
+class SubjectNameStrategy(object):
+    """
+    Instructs the Serializer on how to construct subject names when registering
+    schemas. See the Schema Registryy documentation for additional details.
+
+    .. _SchemaRegistry Documentation:
+        https://docs.confluent.io/current/schema-registry/serializer-formatter.html#group-by-topic-or-other-relationships
+
+    """
+
+    # Configuration enum-like structure
+    TopicNameStrategy = 0
+    TopicRecordNameStrategy = 1
+    RecordNameStrategy = 2
+
+    _str = ['TopicNameStrategy', 'TopicRecordNameStrategy', 'RecordNameStrategy']
+
+    def __call__(self, schema, ctx):
+        raise NotImplementedError
+
+    @classmethod
+    def __str__(cls, strategy):
```

Comment:
with `CompatibilityType` you have the constant values as strings rather than a static method like this, and you also have them ALL_CAPS. I suspect the difference is because `CompatibilityType` is used to deconstruct json data from SR. Two points:

1. it seems the capitalization should be the same.
2. would string constant values be better here rather than this method? (i don't know python well enough to know, just asking).